### PR TITLE
Async http client

### PR DESCRIPTION
**Pull request by pdolega**

### CODE REVIEW

File: codebase/base-async/src/main/scala/com/virtuslab/base/async/IdentityHelpers.scala [hunk 1]
```diff
@@ -29,14 +28,16 @@ trait IdentityHelpers extends DefaultJsonProtocol {
...
     val entity = HttpEntity(`application/json`, json)
     val request = HttpRequest(POST, identityUrl)
       .withHeaders(RawHeader("X-Trace-Id", traceId.id))
       .withEntity(entity)
 
-    Http().singleRequest(request).flatMap { response =>
+    identityHttpClient.flatMapRequest(request) { response =>
       if (response.status.isSuccess())
         response.entity
           .dataBytes
```
Comment (lbialy) [inline codebase/base-async/src/main/scala/com/virtuslab/base/async/IdentityHelpers.scala]: Isnt this going to fail? You are discarding entity bytes before this happens, aren't you?

Comment (pdolega) [reply codebase/base-async/src/main/scala/com/virtuslab/base/async/IdentityHelpers.scala]: No, I don't think so, at least that was not my intent. 

```
    akkaHttp
      .singleRequest(httpRequest)
      .map { response =>
        val result = logic(response) // <--- here i take response and process it (syncronously)
        response.discardEntityBytes() // <--- here i discard
        result 
      }
```

Comment (pdolega) [reply codebase/base-async/src/main/scala/com/virtuslab/base/async/IdentityHelpers.scala]: But then again - i might have just missed something obvious as i did all this change while participating in the call so i might have been sloppy somewhere (and I am still on the call :/ )

Comment (lbialy) [reply codebase/base-async/src/main/scala/com/virtuslab/base/async/IdentityHelpers.scala]: We'll see if that helps then.

**Pull request merged**